cassiopeia/data/CassiopeiaTree.py

```diff
@@ -124,8 +125,9 @@ def __init__(
         self.__network = None
         self.__cache = {}
 
-        self.__original_character_matrix = None
-        self.__current_character_matrix = None
+        self._layers = Layers(self, None)
+
+        self._character_matrix = None
```

Comment:
Do you think it would be useful to have some kind of function called `freeze_character_matrix`, which freezes the current dataframe in `cassiopeia_tree.character_matrix` into some layer?
(of course, this can be done manually by setting `cassiopeia_tree.layers['layer_name'] = cassiopeia_tree.character_matrix.copy()`)

Reply:
I like this idea -- I'll add this functionality. Thanks for the suggestion!

Reply:
Implemented in latest commit.